community/sample-apps/nextjs-postgres-graphql/README.md

```diff
@@ -30,81 +33,116 @@ Boilerplate to get started with Nextjs, Hasura GraphQL engine as CMS and postgre
   ![Insert data into author table](../gatsby-postgres-graphql/assets/browse_rows.jpg)
 
 - Clone this repo:
+
   ```bash
   git clone https://github.com/hasura/graphql-engine
   cd graphql-engine/community/sample-apps/nextjs-postgres-graphql
   ```
 
 - Install npm modules:
+
   ```bash
   npm install
   ```
 
-- Create config.js as follows, in this step we are configuring `withData` with an `httpLink` to connect to a valid GraphQL server URL.
-  ```js
-  import { withData } from 'next-apollo'
-  import { HttpLink } from 'apollo-link-http'
+- Rename the `.env.example` to `.env` and set the following variables
+
+  - `NEXT_PUBLIC_HASURA_APP_URL`:- to your GraphQL API URL
+  - `NEXT_PUBLIC_HASURA_ADMIN_SECRET`:- to your Hasura Admin Secret
+
+- Create config.js as follows :
+  - In the first step, we initialize ApolloClient by passing its constructor a configuration object with the `uri`, `cache`, and `headers`.
+
+  - In the second step, we create the HOC component using the `withApollo` function that takes 2 inputs:
+    1) a callback that returns the ApolloClient object.
+    1) an object with a prop: `render` - used to wrap **your** pages with `<ApolloProvider>`
+
+  ```jsx
+
+  import withApollo from 'next-with-apollo';
+  import { ApolloClient, ApolloProvider, InMemoryCache} from '@apollo/client';
+
+  // creating the Apollo Client
+  const client = new ApolloClient({
+        headers: {
+          'x-hasura-admin-secret': process.env.NEXT_PUBLIC_HASURA_ADMIN_SECRET
```

Comment:
Same here. Admin secret should never be used on the client.

Reply:
Done 👍